src/forms/services/form-submission.service.ts

```
@@ -1559,61 +1596,82 @@
-    const pages: Record<string, any> = {};
-    const formData: Record<string, any> = {};
 
-    // Build fieldId to pageName and fieldName mappings from schema
+      // Filter updatedFieldValues to only include form fields (not custom fields)
+      // Get all field IDs that are part of the current form schema
+      const formFieldIds = new Set<string>();
     const fieldIdToPageName: Record<string, string> = {};
     const fieldIdToSchemaFieldName: Record<string, string> = {};
 
+      // Debug: Log the schema structure to understand the dependencies
+      LoggerUtil.warn(`Schema structure (buildApplicationFromDB): ${JSON.stringify(schema, null, 2)}`);
```

ESLint

[error] Replace ``Schema·structure·(buildApplicationFromDB):·${JSON.stringify(schema,·null,·2)}`);` with `⏎········`Schema·structure·(buildApplicationFromDB):·${JSON.stringify(⏎··········schema,⏎··········null,⏎··········2⏎········)}``

(prettier/prettier)